packages/common/src/eips/2930.json

```diff
@@ -0,0 +1,22 @@
+{
+    "name": "EIP-2929",
+    "comment": "Optional access lists",
+    "url": "https://eips.ethereum.org/EIPS/eip-2930",
+    "status": "Draft",
+    "minimumHardfork": "berlin",
+    "gasConfig": {},
+    "gasPrices": {
+      "accessListStorageKeyCost": {
+        "v": 1900,
+        "d": "Gas cost per storage key in an Access List transaction"
+      },
+      "accessListAddressCost": {
+        "v": 2400,
+        "d": "Gas cost per storage key in an Access List transaction"
+      }
+    },
+    "requiredEIPs": [2718, 2929],
```

Comment:
Some nit: such a field such be defined higher in the parameter list - after `minimumHardfork` e.g., and not between this gas and config parameter dictionaries.

Apart from that: such a parameter has the problem that this doesn't include (at least in the implementation presented here) EIPs included in (encapsulated by) hardforks. So this only works if the dependent EIPs are then at some point included in the same HF.

I would prefer if we give this some more thought before we introducing such an option here (or maybe we directly find the solution along this discussion), so that this would also work if e.g. the dependent EIPs are added to a separate HF (so here: 2718, 2929).

This whole question here in the library of the relation of HFs and EIPs is generally a complex one and needs some careful thought.

Reply:
(sorry, I know this is a bit a side track and you are interested in more general feedback 😄 , just since we are already on it)

Update: so this is highly connected to the two HF file formats with the new format now being used for the first time for [berlin](https://github.com/ethereumjs/ethereumjs-monorepo/blob/master/packages/common/src/hardforks/berlin.json) and allowing to destructure the hardfork files and place the parameters in the respective EIP files (instead of clustering them in the HF file).

Maybe we'll target to continue this for the other HFs relatively soon, since this is generally the cleaner solution and allows for more flexibility in using this library? 🤔  For this use case here we can then also iterate through the HF definitions and see if the EIPs being required are present *either* in the `eips` parameter *or* in one of the active HFs?

With this perspective ahead I would say we can then also eventually leave the implementation here since - in this case - all EIPs affected will be integrated in the same HF. And latest with including the final version of `berlin` we can do this HF file decomposition.

Thoughts?